File: app/src/main/java/com/example/movies/ReviewAdapter.java
package com.example.movies;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;


import java.util.ArrayList;
import java.util.List;

public class ReviewAdapter extends RecyclerView.Adapter<ReviewAdapter.ViewHolder> {

    private List<Review> reviewList = new ArrayList<>();

    private final static String TYPE_POSITIVE = "Позитивный";
    private final static String TYPE_NEUTRAL = "Нейтральный";
    private final static String TAG = "ReviewAdapter";

    public void setReviewList(List<Review> reviewList) {
        this.reviewList = reviewList;
        //Из-за него я потратил час
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.review_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Review review = reviewList.get(position);

        String type = review.getType();
        int backgroundId = android.R.color.holo_red_light;;
        if(type != null) {
            switch (type) {
                case TYPE_POSITIVE:
                    backgroundId = android.R.color.holo_green_light;
                    break;
                case TYPE_NEUTRAL:
                    backgroundId = android.R.color.holo_orange_light;
                    break;
            }
            int backgroundColor = ContextCompat
                    .getColor(holder.itemView.getContext(), backgroundId);
            holder.cardViewBackground.setCardBackgroundColor(backgroundColor);
            holder.textViewReview.setText(review.getReview());
            holder.textViewAuthor.setText(review.getAuthor());
        } else {
            String logError = holder.itemView.getContext().getString(R.string.error);
            String errorApology = holder.itemView.getContext().getString(R.string.error_apology);
            backgroundId = android.R.color.holo_red_dark;
            int backgroundColor = ContextCompat
                    .getColor(holder.itemView.getContext(), backgroundId);

            Log.e(TAG, logError + "\n" + errorApology);

            holder.cardViewBackground.setCardBackgroundColor(backgroundColor);
            holder.textViewAuthor.setText(logError);
            holder.textViewReview.setText(errorApology);
        }
    }

    @Override
    public int getItemCount() {
        return reviewList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView textViewAuthor;
        private final TextView textViewReview;
        private final CardView cardViewBackground;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewAuthor = itemView.findViewById(R.id.textViewAuthor);
            textViewReview = itemView.findViewById(R.id.textViewReview);
            cardViewBackground = itemView.findViewById(R.id.cardViewBackground);
        }
    }
}

File: app/src/main/java/com/example/movies/adapters/ReviewAdapter.kt
package com.example.movies.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.example.movies.databinding.ReviewItemBinding
import com.example.movies.pojo.Review

class ReviewAdapter : RecyclerView.Adapter<ReviewAdapter.ViewHolder>() {

    var reviewList: List<Review> = mutableListOf()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    companion object {
        private const val TYPE_POSITIVE = "Позитивный"
        private const val TYPE_NEUTRAL = "Нейтральный"
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = ReviewItemBinding.inflate(
            LayoutInflater
                .from(parent.context), parent, false
        )
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val review = reviewList[position]
        val backgroundId = when (review.type) {
            TYPE_POSITIVE -> android.R.color.holo_green_light
            TYPE_NEUTRAL -> android.R.color.holo_orange_light
            else -> android.R.color.holo_red_light
        }
        val backgroundColor = ContextCompat.getColor(holder.itemView.context, backgroundId)
        with(holder.binding) {
            cardViewBackground.setCardBackgroundColor(backgroundColor)
            textViewReview.text = review.review
            textViewAuthor.text = review.author
        }
    }

    override fun getItemCount() = reviewList.size

    inner class ViewHolder(val binding: ReviewItemBinding) : RecyclerView.ViewHolder(binding.root)
}